[NUI] Enhance Text documents (#1428)
Samsung/tizen-docs: docs/application/dotnet/guides/nui/text.md

    ![Text with color underline](media/TextWithColorUnderline.png)

> -   To outline the text label, set the `Outline` property::

Author: Done

    ```csharp
    TextLabel label = new TextLabel("Text with Color Outline");
    label.TextColor = Color.White;
    PropertyMap outline = new PropertyMap();
    outline.Add("color", new PropertyValue(Color.Red));
    outline.Add("width", new PropertyValue(2.0f)); /// 2-pixel width
    label.Outline = outline;
    Window.Instance.Add(label);
    ```

>     **Figure: Text with red outline**

Author: Hi @annie-samsung ,

If I add more white space, will there be more space?
I don't think so.

If there is 1 white space or 2 more spaces, the result looks the same.
Could you please tell me how? then I will update it.

window.Add(field);
```

When the `TextField` is tapped, it automatically gets the keyboard focus. Key events correspond to entering the text. Additionally, the placeholder text is removed as soon as the text is entered. The text entered can be retrieved by using the `Text` property:

```csharp
string fieldTextString = field.Text;
```

> ### Use Placeholder Property in TextField

Reviewer: Global: 
Can we use the Sentence case for all headings? Let me know if that is possible.

Author: I am sorry, what is Sentence case?

Author: Class or property names cannot be written in lowercase.
In other cases I can change it.

For example :
### Use Placeholder Property in TextField -> ### Use Placeholder property in TextField.
(Placeholder = property, TextField = class)

### Align Text in TextField -> ### Align text in TextField
### Use HiddenInputSettings Property in TextField -> ### Use HiddenInputSettings property in TextField

Please let me know if you want this change for all headings.

Reviewer: Oh sorry! I didn't mean that exactly.

What my intention was that heading of this page which starts with (#, ##, ###) are, written in Title case (we capitalize ist letter of each word in a title or heading)

I suggested using sentence case for headings, which means:

In sentence case, lowercase most words in a title or heading. Capitalize only the Ist letter of the sentence: 

e.g 

### Use Placeholder Property  (Title case)

### Use placeholder property (Sentence case) 

If there is a name in the sentence case, we write that as capital.

### Use placeholder property in TextField (correct)
### Use placeholder property in textfield (Incorrect)

Author: Thanks for your explanation.

By the way, in below case,
### Use placeholder property in TextField (correct)

"Placeholder" is the name of the property(attribute).
So, I think below is correct.
What do you think?
### Use Placeholder property in TextField

field.HorizontalAlignment = HorizontalAlignment.Begin;
```

### Use HiddenInputSettings property in TextField

The `TextField` class can substitute the input text using the `HiddenInputSettings` property.

> To set the `HiddenInputSettings` for detail options, you can use [HiddenInputProperty](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.HiddenInputProperty.html) and [HiddenInputModeType](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.HiddenInputModeType.html).

Reviewer: Just one doubt :  To set the `HiddenInputSettings` for detailed options, 
What are the detailed options? 

I am not able to get this, can you elaborate. 
```suggestion
To set the `HiddenInputSettings` for detailed options, you can use [HiddenInputProperty](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.HiddenInputProperty.html) and [HiddenInputModeType](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.HiddenInputModeType.html).

Author: @nawab-samsung 

The detailed options are Mode and Properties of HiddenInputSettings.
(Struct HiddenInputProperty, Enum HiddenInputModeType)

Depending on HiddenInputModeType, TextField behaves as follows.
* HideAll : Hide all the input text.
* HideCount : Hide n characters from start.
* HideNone : Don't hide text.
* ShowCount : Show n characters from start.
* ShowLastCharacter : Show last character for the duration(use ShowLastCharacterDuration property to modify duration).

Also, depending on the mode, through HiddenInputProperty,
ShowLastCharacterDuration, SubstituteCharacter, SubstituteCount values ​​can be set.

I think, users need a detailed explanation of this.
So I made an API reference link for this. (The most detailed explanation)

Reviewer: @wonrst thanks.

> To set the `HiddenInputSettings` for detailed options, you can use [HiddenInputProperty](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.HiddenInputProperty.html) and [HiddenInputModeType](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.HiddenInputModeType.html).

Reviewer 2: Hello, @annie-samsung 

As I commented on messenger, would you please check the links to API here?

.NET guide are updated by https://github.com/Samsung/tizen-docs/pull/1445,
and one of the update is changing the links to API.

So I think these links should be checked with author of https://github.com/Samsung/tizen-docs/pull/1445
whether if it is ok to use https://samsung.github.io/ link.

Or, another temporary solution in my thought is,
removing the links to `HiddenInputProperty` and `HiddenInputModeType` for now, and adding it back later.

Would you please check?

Reviewer 3: @mijin85cho , sure will sort this out

Reviewer 3: @piotrekgorny, in this PR you can see that in text.md file at line 456 github.io links are provided, could you suggest how do we go about with this PR, do we remove them, an alternate link is available?

@wonrst, Piotr's team just finished a big exercise of removing github.io links from all NUI guides in PR
https://github.com/Samsung/tizen-docs/pull/1387, so adding it back via this PR can cause problem.

> To set the `HiddenInputSettings` for detailed options, you can use [HiddenInputProperty](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.HiddenInputProperty.html) and [HiddenInputModeType](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.HiddenInputModeType.html).

Reviewer 3: ```suggestion
To set the `HiddenInputSettings` for detailed options, you can use [HiddenInputProperty](/application/dotnet/api/TizenFX/latest/api/Tizen.NUI.HiddenInputProperty.html) and [HiddenInputModeType](/application/dotnet/api/TizenFX/latest/api/Tizen.NUI.HiddenInputModeType.html).

```

};
```

### Decoration of Grab Handle and Selection Handle

> To decorate the grab handle and selection handle, you can use the `GrabHandleImage`, `SelectionHandleImageLeft`, `SelectionHandleImageRight` properties.

Reviewer: ```suggestion
To decorate the grab handle and selection handle, you can use the `GrabHandleImage`, `SelectionHandleImageLeft`, and `SelectionHandleImageRight` properties.

Author: Done

> Also you can change the color of handle using `GrabHandleColor` property.

Reviewer: ```suggestion
Also, you can change the color of the handle using the `GrabHandleColor` property.

Author: Done

Window.Instance.Add(editor);
```

> ### TextEditor properties

Reviewer: Line 682 
## Related information

Author: Done

| `TranslatableText`                 | String      | Specifies the `TranslatableText` property that sets the SID value. |

## Related information
- Examples

>   - [TvTextSample](https://github.com/Samsung/Tizen-CSharp-Samples/tree/master/TV/TextSample)

Author: Done
